cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureTerminationHelperService.java

```diff
@@ -95,18 +95,31 @@
         deleteVolumeSets(ac, stack, resourcesToRemove, networkResources, resourceGroupName);
 
         if (deleteWholeDeployment) {
+            // deleting availability sets
             List<String> availabiltySetNames = getResourceNamesByResourceType(resourcesToRemove, ResourceType.AZURE_AVAILABILITY_SET);
             azureUtils.deleteAvailabilitySets(client, resourceGroupName, availabiltySetNames);
             deleteCloudResourceList(ac, resourcesToRemove, ResourceType.AZURE_AVAILABILITY_SET);
 
+            // deleting networks
             List<String> networkIds = getResourceIdsByResourceType(resourcesToRemove, ResourceType.AZURE_NETWORK);
             azureUtils.deleteNetworks(client, networkIds);
             deleteCloudResourceList(ac, resourcesToRemove, ResourceType.AZURE_NETWORK);
             deleteCloudResourceList(ac, resourcesToRemove, ResourceType.AZURE_SUBNET);
 
+            // deleting security groups
             List<String> securityGroupIds = getResourceIdsByResourceType(resourcesToRemove, ResourceType.AZURE_SECURITY_GROUP);
             azureUtils.deleteSecurityGroups(client, securityGroupIds);
             deleteCloudResourceList(ac, resourcesToRemove, ResourceType.AZURE_SECURITY_GROUP);
+
+            // deleting images
+            List<String> imageIds = getResourceIdsByResourceType(resourcesToRemove, ResourceType.AZURE_MANAGED_IMAGE);
+            azureUtils.deleteImages(client, imageIds);
+            deleteCloudResourceList(ac, resourcesToRemove, ResourceType.AZURE_MANAGED_IMAGE);
+
+            // deleting storage account
+            List<String> accountIds = getResourceIdsByResourceType(resourcesToRemove, ResourceType.AZURE_STORAGE);
+            azureUtils.deleteStorageAccounts(client, accountIds);
+            deleteCloudResourceList(ac, resourcesToRemove, ResourceType.AZURE_STORAGE);
```

Comment:
I am not convinced this is the right place. What if a datahub is being terminated, will it delete the images and the storage account? What about concurrent termination of cluster A and provisioning of cluster B? I might be missing something, sorry for that.

Reply:
We had a conversation with @pdarvasi. This would cause a performance hit in case a customer is creating a large number of short-lived datahubs: the image would be copied frequently, causing 10-20 minutes of extra datahub provisioning time. If this potential performance hit is not a problem then this PR can be merged.

Reply:
As clarified, it will not cause any perf hit.